common/achievement/get.ts

```diff
@@ -156,37 +156,32 @@ const assembleAchievementData = async (userAchievements: achievements_with_templ
 
     let maxValue: number = achievementTemplates.length;
     let currentValue: number = currentAchievementIndex;
-    if (
-        userAchievements[currentAchievementIndex].template.type === AchievementType.STREAK ||
-        userAchievements[currentAchievementIndex].template.type === AchievementType.TIERED
-    ) {
-        const dataAggregationKeys = Object.keys(userAchievements[currentAchievementIndex].template.conditionDataAggregations as Prisma.JsonObject);
+    if (userAchievements[currentAchievementIndex].template.type === AchievementType.STREAK) {
         const evaluationResult = await isAchievementConditionMet(userAchievements[currentAchievementIndex]);
-        if (evaluationResult) {
-            currentValue = dataAggregationKeys.map((key) => evaluationResult.resultObject[key]).reduce((a, b) => a + b, 0);
-            maxValue =
-                userAchievements[currentAchievementIndex].template.type === AchievementType.STREAK
-                    ? userAchievements[currentAchievementIndex].recordValue !== null && userAchievements[currentAchievementIndex].recordValue > currentValue
-                        ? userAchievements[currentAchievementIndex].recordValue
-                        : currentValue
-                    : dataAggregationKeys
-                          .map((key) => {
-                              // TODO: check if we can remove valueToAchieve
-                              return Number((userAchievements[currentAchievementIndex].template.conditionDataAggregations as any)[key].valueToAchieve || 0);
-                          })
-                          .reduce((a, b) => a + b, 0);
-            if (
-                userAchievements[currentAchievementIndex].template.type === AchievementType.STREAK &&
-                currentValue < maxValue &&
-                userAchievements[currentAchievementIndex].achievedAt
-            ) {
-                await prisma.user_achievement.update({
-                    where: { id: userAchievements[currentAchievementIndex].id },
-                    data: { achievedAt: null, isSeen: false },
-                });
-            }
+        currentValue = evaluationResult.aggregationResult;
+        maxValue = userAchievements[currentAchievementIndex].recordValue;
```

Comment:
What is recordValue for non-TIERED? It's a bit confusing that it is reassigned below.

Reply:
🤔  The record value is only used for streaks, but the maxValue will be used differently based on the type.

Tiered maxValue => How many events do you need to achieve? Like: 1 / 3 Wochen gelernt.
Streak maxValue => The record value
Sequential maxValue => How many steps do we have in the sequence.

Reply:
Is this documented somewhere?

Reply:
I'll start writing the texts when everything is live